### PR TITLE
add authentication, blobstore and FindMeetingQuery(week4)

### PR DESCRIPTION
**Pull request by hzw649**

### CODE REVIEW

**Comment by akovski** [comment]
Do you have accessible url for your site?

**Comment by hzw649** [comment]
https://zhu-sps-summer20.df.r.appspot.com
just complete basic features

File: portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+ * When the user submits the form, Blobstore processes the file upload and then forwards the request
+ * to this servlet. This servlet can then process the request using the file URL we get from
+ * Blobstore.
+ */
+@WebServlet("/my-form-handler")
+public class FormHandlerServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
```
Comment (akovski) [inline portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java]: clear blank in the beginning of function body

File: portfolio/src/main/java/com/google/sps/servlets/LoginStateServlet.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+@WebServlet("/login-state")
+public class LoginStateServlet extends HttpServlet {
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        PrintWriter out = response.getWriter();
+        UserService userService = UserServiceFactory.getUserService();
+
+        int state = 0;
```
Comment (akovski) [inline portfolio/src/main/java/com/google/sps/servlets/LoginStateServlet.java:43]: Add comment for meaning of state's value

File: portfolio/src/main/java/com/google/sps/servlets/LoginStateServlet.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        String url = "";
+        // If user is not logged in, show a login form (could also redirect to a login page)
+        if (userService.isUserLoggedIn())
+        {
+            state = 1;
+            url = userService.createLogoutURL("/");
+        }
+        else
+            url = userService.createLoginURL("/");
+        String json = "{ \"state\":"+state;
```
Comment (akovski) [inline portfolio/src/main/java/com/google/sps/servlets/LoginStateServlet.java]: this looks very fragile, is there a library to generate json string?

File: portfolio/src/main/java/com/google/sps/servlets/LoginStateServlet.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        PrintWriter out = response.getWriter();
+        UserService userService = UserServiceFactory.getUserService();
+
+        int state = 0;
+        String url = "";
+        // If user is not logged in, show a login form (could also redirect to a login page)
+        if (userService.isUserLoggedIn())
```
Comment (akovski) [inline portfolio/src/main/java/com/google/sps/servlets/LoginStateServlet.java:46]: please keep consistent style of brackets

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -13,6 +13,31 @@
 // limitations under the License.
 
 function LoadData() {
+    fetch('/login-state').then(rsp => rsp.json()).then((stat) => {
+        const commentElement = document.getElementById('comment-list');
+        const loginElement = document.getElementById('login-link');
+        console.log(stat);
```
Comment (akovski) [inline portfolio/src/main/webapp/script.js]: clear debugging code

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 6]
```diff
@@ -58,11 +57,14 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
         String title = request.getParameter("comment");
         long timestamp = System.currentTimeMillis();
+        String email = UserServiceFactory.getUserService().getCurrentUser().getEmail();
 
         Entity taskEntity = new Entity("Comments");
         taskEntity.setProperty("title", title);
         taskEntity.setProperty("timestamp", timestamp);
```
Comment (akovski) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: clear unnecessary blank line

File: portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    // Get the message entered by the user.
+    String message = request.getParameter("message");
+
+    // Get the URL of the image that the user uploaded to Blobstore.
```
Comment (akovski) [inline portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java]: very detailed comment! It's good to have comment, personally I feel it's not necessarry to add comment if the code logic or variable is obvious enough.

**Review by akovski — COMMENTED** [review]
Sorry for delayed review.

is FindMeettingQuery relevant? I think it can be splited into a seperate pull request.

authentication and blobstore functionalities can also be splited but I'm not requesting it since we are late this week.

**Comment by hzw649** [comment]
updated @akovski

**Comment by akovski** [comment]
Hi, you seem not update other code with comment inside

**Comment by hzw649** [comment]
sorry, I have checked code in FormHandlerServlet.java and delete unnecessary comment

**Pull request merged**